src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs

```diff
@@ -3038,6 +3038,48 @@ public void TestAttachmentIf()
             TestSimpleBindingSuccess("If(true, DS, DS)", schema, symbol);
         }
 
+        [Theory]
+        [InlineData("X")]
+        [InlineData("$\"Test {X}\"")]
+        [InlineData("!X")]
+        [InlineData("Table(X)")]
+        [InlineData("X.Field1")]
+        [InlineData("X + 1")]
+        [InlineData("X + Date(2022, 10, 27)")]
+        [InlineData("X * 1")]
+        [InlineData("X < 2")]
+        [InlineData("\"The\" in X")]
+        [InlineData("123 in X")]
+        [InlineData("123 in Table(X)")]
+        [InlineData("IsBlank(X)")]
+        [InlineData("IsError(X)")]
+        [InlineData("Text(X)")]
+        [InlineData("Value(X)")]
+        [InlineData("Boolean(X)")]
+        [InlineData("Index([1,2,3], X)")]
+        public void DeferredTypeTest(string script)
+        {
+            Preview.FeatureFlags.StringInterpolation = true;
+            var symbolTable = new SymbolTable();
+            symbolTable.AddVariable("X", FormulaType.Unknown);
+            TestDeferredTypeBindingWarning(script, symbolTable);
+        }
+
+        [Theory]
+        [InlineData("$\"Test {X} {XR}\"")]
+        [InlineData("Table(X, XN)")]
+        [InlineData("X.field + XN.field")]
+        [InlineData("Index([1,2,3], X).missing")]
+        public void DeferredTypeTest_Negative(string script)
+        {
+            Preview.FeatureFlags.StringInterpolation = true;
+            var symbolTable = new SymbolTable();
+            symbolTable.AddVariable("X", FormulaType.Unknown);
+            symbolTable.AddVariable("XN", FormulaType.Number);
```

Comment:
>XN [](http://example.com/codeflow?start=37&length=2)

minor nit: for convention in these tests, we could have X be the Unknown type. 

For the non-deferred types, can we avoid the 'X' prefix - we call them "N" or "R"